File: Library/src/main/java/org/endeavourhealth/library/model/imq/ArgumentReference.java
package org.endeavourhealth.library.model.imq;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.endeavourhealth.library.model.tripletree.TTIriRef;

@Getter
@NoArgsConstructor
@AllArgsConstructor
public class ArgumentReference {
  private String parameter;
  private TTIriRef referenceIri;
  private TTIriRef dataType;

  public ArgumentReference setParameter(String parameter) {
    this.parameter = parameter;
    return this;
  }

  public ArgumentReference setReferenceIri(TTIriRef referenceIri) {
    this.referenceIri = referenceIri;
    return this;
  }

  public ArgumentReference setDataType(TTIriRef dataType) {
    this.dataType = dataType;
    return this;
  }
}

File: src/interfaces/ArgumentReference.ts
import z from "zod";

import { TTIriRef, TTIriRefSchema } from "./TTIriRef";

// export interface ArgumentReference {
//   parameter?: string;
//   referenceIri?: TTIriRef;
//   dataType?: TTIriRef;
// }

export const ArgumentReferenceSchema = z.strictObject({
  parameter: z.string().optional(),
  referenceIri: TTIriRefSchema.optional(),
  dataType: TTIriRefSchema.optional()
});

export type ArgumentReference = z.output<typeof ArgumentReferenceSchema>;
